engine/internal/runtime/manager.go

```diff
@@ -278,14 +294,23 @@ func (m *Manager) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result,
 		// This would call when the manager synchronizes the cache
 		// for the first time or when another engine creates a runtime.
 		log.V(4).Info("Registering runtime", "model", modelID)
-		if added, err := m.addRuntime(modelID, sts); err != nil {
+		if ready, added, err := m.addRuntime(modelID, sts); err != nil {
 			log.Error(err, "Failed to add runtime")
 			return ctrl.Result{}, err
 		} else if added {
 			if err := m.autoscaler.Register(ctx, modelID, &sts); err != nil {
 				log.Error(err, "Failed to register autoscaler")
 				return ctrl.Result{}, err
 			}
+			if !ready {
+				if yes, err := hasUnschedulableChildren(ctx, m.k8sClient, sts); err != nil {
```

Comment:
Question: Suppose that there are two pods for the statefulset, and one of the pods is unschedulable. In such a case, should we still cancel the request?

Reply:
If so, shouldn't we change `hasUnschedulableChildren` so that it returns true only if all pods are unschedulable?

Reply:
It's unnecessary, but changing it for better readability would be fine.

Reply:
ah i see what you meant. No strong opinion!